.github/workflows/publish.yaml

```diff
@@ -0,0 +1,34 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  charm:
+    name: Charm
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: |
+        sudo snap install charm --classic
+        sudo pip3 install charmcraft
+
+    - name: Publish bundle
+      env:
+        CHARMSTORE_CREDENTIAL: ${{ secrets.CHARMSTORE_CREDENTIAL }}
+      run: |
+        set -eux
+        echo $CHARMSTORE_CREDENTIAL > ~/.go-cookies
+        IMAGE_TAG="v2.28.1"
```

Comment:
Can this get updated to use `yq`, like this?
https://github.com/canonical/spark-operator/blob/master/.github/workflows/cd.yaml

I think reading it from metadata.yaml will work better than copying the version here